ecs-cli/modules/cli/local/localproject/project.go

```diff
@@ -200,68 +202,91 @@ var readTaskDefFromRemote = func(remote string, p *localProject) (*ecs.TaskDefin
 
 // Convert translates an ECS Task Definition into a Compose V3 schema and
 // stores the data on the project
-func (p *localProject) Convert() error {
+func (p *LocalProject) Convert() error {
+	if err := p.convertBaseCompose(); err != nil {
+		return err
+	}
+	if err := p.convertOverrideCompose(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *LocalProject) convertBaseCompose() error {
 	conf, err := converter.ConvertToComposeConfig(p.taskDefinition, p.inputMetadata)
 	if err != nil {
 		return err
 	}
-	data, err := converter.MarshalComposeConfig(*conf, LocalOutDefaultFileName)
+	data, err := converter.MarshalComposeConfig(*conf, p.LocalOutFileName())
 	if err != nil {
 		return err
 	}
+	p.baseCompose = data
+	return nil
+}
 
-	p.localBytes = data
+func (p *LocalProject) convertOverrideCompose() error {
+	conf, err := converter.ConvertToComposeOverride(p.taskDefinition)
+	if err != nil {
+		return err
+	}
+	data, err := converter.MarshalComposeConfig(*conf, p.OverrideFileName())
+	if err != nil {
+		return err
+	}
+	p.overrideCompose = data
 	return nil
 }
 
 // Write writes the compose data to a local compose file. The output filename is stored on the project
-func (p *localProject) Write() error {
-	// Will error if the file already exists, otherwise create
-	p.localOutFileName = LocalOutDefaultFileName
+func (p *LocalProject) Write() error {
+	if err := p.writeFile(p.LocalOutFileName(), p.baseCompose); err != nil {
+		return err
+	}
+	logrus.Infof("Successfully wrote %s", p.LocalOutFileName())
 
-	if fileName := p.context.String(flags.Output); fileName != "" {
-		p.localOutFileName = fileName
+	if err := p.writeFile(p.OverrideFileName(), p.overrideCompose); err != nil {
+		return err
 	}
+	logrus.Infof("Successfully wrote %s", p.OverrideFileName())
 
-	return p.writeFile()
+	return nil
 }
 
-func (p *localProject) writeFile() error {
-	out, err := openFile(p.localOutFileName)
+func (p *LocalProject) writeFile(filename string, content []byte) error {
+	out, err := openFile(filename)
 	defer out.Close()
 
 	// File already exists
 	if err != nil {
-		return p.overwriteFile()
+		return p.overwriteFile(filename, content)
 	}
 
-	_, err = out.Write(p.localBytes)
+	_, err = out.Write(content)
 
 	return err
 }
 
 // Facilitates test mocking
-var openFile func(filename string) (*os.File, error) = func(filename string) (*os.File, error) {
+var openFile = func(filename string) (*os.File, error) {
```

Comment:
Just curious why do we want `openFile` to be assigned a function handle instead of using it as a function name?

Reply:
I believe it's so that it can be changed for unit tests https://github.com/aws/amazon-ecs-cli/blob/ecs-local/ecs-cli/modules/cli/local/localproject/project_test.go#L156-L164

Reply:
I see...

Reply:
For my own edification, is there a reason to remove the type signature on the variable declaration? Is it just redundant for anonymous functions?

Reply:
I believe it's redundant if an initializer is present https://github.com/aws/amazon-ecs-cli/pull/854#discussion_r300733974